BuildingThemes/GUI/UIThemeManager.cs

```diff
@@ -0,0 +1,563 @@
+            {
+                // Catching any exception to not block the loading process of other mods
+                Debugger.Log("Building Themes: An error has happened during the UI creation.");
+                Debugger.LogException(e);
+            }
+        }
+
+        public static void Destroy()
+        {
+            try
+            {
+                if (_gameObject != null)
+                    GameObject.Destroy(_gameObject);
+            }
+            catch (Exception e)
+            {
+                // Catching any exception to not block the unloading process of other mods
+                Debugger.Log("Building Themes: An error has happened during the UI destruction.");
+                Debugger.LogException(e);
+            }
+        }
+
+        public void Toggle()
+        {
+            if (isVisible)
+            {
+                Hide();
+            }
+            else
+            {
+                base.Show();
+
+                if (m_themeSelection.selectedIndex == -1) m_themeSelection.selectedIndex = 0;
+            }
+        }
+
+        public void CreateTheme(string name)
+        {
+            if (BuildingThemesManager.instance.GetThemeByName(name) == null)
+            {
+                Configuration.Theme newTheme = new Configuration.Theme(name);
+
+                BuildingThemesManager.instance.Configuration.themes.Add(newTheme);
+                m_isDistrictThemesDirty = true;
+
+                InitBuildingLists();
+
+                m_themeSelection.rowsData.m_buffer = m_allThemes;
+                m_themeSelection.rowsData.m_size = m_allThemes.Length;
+
+                for (int i = 0; i < m_allThemes.Length; i++)
+                {
+                    if (m_allThemes[i] == newTheme)
+                    {
+                        m_themeSelection.DisplayAt(i);
+                        m_themeSelection.selectedIndex = i;
+                    }
+                }
+            }
+        }
+
+        public void DeleteTheme(Configuration.Theme theme)
+        {
+            if (!theme.isBuiltIn)
+            {
+                BuildingThemesManager.instance.Configuration.themes.Remove(theme);
+                m_isDistrictThemesDirty = true;
+
+                InitBuildingLists();
+
+                m_themeSelection.rowsData.m_buffer = m_allThemes;
+                m_themeSelection.rowsData.m_size = m_allThemes.Length;
+                m_themeSelection.DisplayAt(0);
+                m_themeSelection.selectedIndex = 0;
+            }
+        }
+
+        public void ChangeBuildingStatus(BuildingItem item, bool include)
+        {
+            if (item.building != null && item.building.isBuiltIn)
+            {
+                item.building.include = include;
+            }
+            else if (include)
+            {
+                Configuration.Building building = new Configuration.Building(item.name);
+
+                if (!selectedTheme.buildings.Contains(building))

```

Comment:
I think this doesn't work because no Equals or HashCode method is set up.

Better do ``selectedTheme.containsBuilding(building.name);``